### ISSUE TITLE
[ttnn.concat]: produces garbage values when concating tensors of shape `[32, 1, 3, 8192]` and `[32, 1, 512, 8192]`

### ISSUE THREAD

**Issue by lingjunwenTT** [tenstorrent/tt-metal#47293]
### Component / Area

ops, tt-metal

### Issue Type (optional)

Bad Outputs

### Observed

Concating two tensors of shape `[32, 1, 3, 8192]` and `[32, 1, 512, 8192]` produces gargabe values when in `TILE_LAYOUT`.

```
ttnn.concat([pad[B, 1, 3, 8192], x[B, 1, 512, 8192]], dim=2) 

 batch      layout   PCC vs torch   max|err|   verdict
--------------------------------------------------------------
    32        TILE         0.0000      8.303   *** MISMATCH (garbage) ***
    32   ROW_MAJOR         1.0000      0.016   OK
```

### Expected

expect the pcc>0.99

### 1. Steps (exact commands)

```python
# SPDX-FileCopyrightText: © 2026 Tenstorrent AI ULC
# SPDX-License-Identifier: Apache-2.0
"""Minimal repro: ttnn.concat on dim=2 is wrong in TILE layout but correct in ROW_MAJOR."""
import torch

import ttnn


def pcc(ref, got):
    a, b = ref.flatten().float(), got.flatten().float()
    return torch.corrcoef(torch.stack([a, b]))[0, 1].item()

def to_tile(t):
    return ttnn.from_torch(t, device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16)

def to_row(t):
    return ttnn.from_torch(t, device=device, layout=ttnn.ROW_MAJOR_LAYOUT, dtype=ttnn.bfloat16)

device = ttnn.open_mesh_device(mesh_shape=ttnn.MeshShape(1, 1))
a = torch.randn(32, 1, 3, 8192)
b = torch.randn(32, 1, 512, 8192)
ref = torch.cat([a, b], dim=2)

a_tile, b_tile = to_tile(a), to_tile(b)
a_row, b_row = to_row(a), to_row(b)

concat_tile = ttnn.to_torch(ttnn.concat([a_tile, b_tile], dim=2))
concat_row = ttnn.to_torch(ttnn.concat([a_row, b_row], dim=2))

print(f"TILE      PCC vs torch: {pcc(ref, concat_tile):.4f}") # THIS IS A LOW NUMBER!
print(f"ROW_MAJOR PCC vs torch: {pcc(ref, concat_row):.4f}")

ttnn.close_mesh_device(device)
```

### 2. Input data / link or description

`torch.rand` is fine.

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 24.04

<img width="836" height="168" alt="Image" src="https://github.com/user-attachments/assets/c407a4d5-49ea-4f34-a021-0e91f6e4d1a3" />

### 2. Hardware Details

Blackhole QB2

### Is this a regression?

Unknown

### Priority

P1

### Impact

Models that require concatentation are blocked and we'll need to switch to ROW layout!

**Comment by ronnie00TT** [comment]
@cmaryanTT @bgoelTT @lingjunwenTT Follow-up in PR [#47470](https://github.com/tenstorrent/tt-metal/pull/47470)

**Comment by ronnie00TT** [comment]
Fixed in PR #47516 , close this issue for now.

**Comment by minaliuTT** [comment]
@lingjunwenTT @ronnie00TT Did this get closed as P0 or P1?

**Comment by bgoelTT** [comment]
This issue was made before the Priority labels were standardized to P0, P1, etc. This should have been closed as P0
